resolvers.ts

```diff
@@ -40,6 +40,38 @@ export const resolvers: Resolvers = {
         };
       }, []);
     },
+    BulkDownloadCSV: async (root, args, context, info) => {
+      const body = {
+        download_type: args?.input?.downloadType,
+        workflow: args?.input?.workflow,
+        params: {
+          include_metadata: { value: true }, 
+        sample_ids: {
+          value: args?.input?.workflowRunIds
+        }, 
+        workflow: {
+          value: "consensus_genome"
+        }
+        },
+        workflow_run_ids: args?.input?.workflowRunIds,
+      };
+      const res = await postWithCSRF(
+        `/bulk_downloads`,
+        body,
+        args,
+        context,
+        true
+      );
+      if (res.status !== 200){
+        return {
+          url: null,
+          error: res.statusText,
+        }
+      }
+      return {
+        url: res.csvUrl ? res.csvUrl : "success",
+      }
```

Comment:
I was planning to send the file back directly, using [the Rails function 'send_data`](https://api.rubyonrails.org/classes/ActionController/DataStreaming.html#method-i-send_data).  We do this in a few other places in Rails already.  A couple of questions around this though

1.  How is the entities service going to return the file?  A URL only works if we are writing the file somewhere, but my impression is that we weren't.
2. I'm not sure how we represent and federate a file response.  Depending on how that works, do we even need a resolver function for this if we don't need to change the response?  Though I guess we will need it eventually to fetch CG metrics from nextGen and pass them to Rails.

Reply:
For future reference - we decided that we would send back a json string and the front end would be responsible for turning it into a CSV file and downloading.